apps/meteor/app/otr/client/OTRRoom.ts

```diff
@@ -110,6 +112,34 @@ export class OTRRoom implements IOTRRoom {
 		}
 	}
 
+	// Starts listening to other user's status changes and end OTR if any of the Users goes offline
+	// this should be called in 2 places: on acknowledge (meaning user accepted OTR) or on establish (meaning user initiated OTR)
+	listenToUserStatus(userId: IUser['_id']): void {
+		Presence.listen(userId, (event: UserPresence | undefined) => {
+			if (!event) {
+				return;
+			}
+			if (event.status === UserStatus.OFFLINE) {
+				console.warn(`OTR Room ${this._roomId} ended because ${userId} went offline`);
+				this.end();
+				Presence.stop(userId, () => {
```

Comment:
Removing listener won't work this way, as will have to pass the reference of the function which was passed at the time of adding the listener.